Src/Fido2/AuthenticatorResponse.cs

```diff
@@ -68,11 +69,11 @@ protected void BaseVerify(string expectedOrigin, byte[] originalChallenge, byte[
                 throw new Fido2VerificationException("Challenge not equal to original challenge");
 
             var fullyQualifiedOrigin = FullyQualifiedOrigin(Origin);
-            var fullyQualifiedExpectedOrigin = FullyQualifiedOrigin(expectedOrigin);
+            var fullyQualifiedExpectedOrigins = expectedOrigins.Select(FullyQualifiedOrigin);
 
             // 5. Verify that the value of C.origin matches the Relying Party's origin.
-            if (!string.Equals(fullyQualifiedOrigin, fullyQualifiedExpectedOrigin, StringComparison.OrdinalIgnoreCase))
-                throw new Fido2VerificationException($"Fully qualified origin {fullyQualifiedOrigin} of {Origin} not equal to fully qualified original origin {fullyQualifiedExpectedOrigin} of {expectedOrigin}");
+            if (!fullyQualifiedExpectedOrigins.Any(o => string.Equals(fullyQualifiedOrigin, o, StringComparison.OrdinalIgnoreCase)))
```

Comment:
While not a dealbreaker, I'm wondering if a HashSet would be better than a List?
My gut feeling is that it would be able to do a lookup without iterating (like linq Any does).

It is mostly relevant if the list of origins grow due to large number of unique origins.

Reply:
Sounds reasonable, just need to make sure to normalize the ordinal before comparing.

Reply:
I changed it to HashSet, however to avoid iterating and converting them to FullyQualifiedOrigins each time, I had to store them in the `Fido2Configuration` object. I'm not completely happy with it, but I don't see any other place to put it.